### PR TITLE
Made pod-profile-mount-volume service yaml more configurable.

### PR DESCRIPTION
**Pull request by chhsia0**
This PR makes the pod-profile-mount-volume yaml configurable through `service.test_profile_volume` so it can be customized for soaking.

### CODE REVIEW

File: frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml [hunk 1]
```diff
@@ -14,7 +14,12 @@ pods:
       profile:
         goal: RUNNING
         essential: false
-        cmd: df -t $VOLUME_PROFILE pod-container-path >> pod-container-path/output && df -t $VOLUME_PROFILE profile-container-path >> profile-container-path/output && sleep $SLEEP_DURATION
+        cmd: >
```
Comment (elezar) [inline frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml]: Should we start the command with `set -e -x` for easier debugging?
```suggestion
        cmd: >
            set -e -x
```

File: frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml [hunk 2]
```diff
@@ -28,7 +33,12 @@ pods:
       noprofile:
         goal: RUNNING
         essential: false
-        cmd: df -t $VOLUME_PROFILE pod-container-path >> pod-container-path/output && df -t ext4 noprofile-container-path >> noprofile-container-path/output && sleep $SLEEP_DURATION
+        cmd: >
```
Comment (elezar) [inline frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml]:
```suggestion
        cmd: >
            set -e -x
```

File: frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml [hunk 3]
```diff
@@ -14,7 +14,12 @@ pods:
       profile:
         goal: RUNNING
         essential: false
-        cmd: df -t $VOLUME_PROFILE pod-container-path >> pod-container-path/output && df -t $VOLUME_PROFILE profile-container-path >> profile-container-path/output && sleep $SLEEP_DURATION
+        cmd: >
+               [ -z "{{{TEST_PROFILE_VOLUME}}}" ] ||
```
[the remark this answers is not in the document]
Comment (nfnt) [reply frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml]: The change is intended, this is simply testing if we are able to write (something) into `output`. Let's not use conditionals here as we want to fail early. I.e. if `TEST_PROFILE_VOLUME_COMMAND` isn't set, the task should fail.

File: frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml (hunk 1 above)
Comment (nfnt) [reply frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml]: 👍

File: frameworks/helloworld/universe/config.json [hunk 4]
```diff
@@ -58,6 +58,11 @@
           "type": "boolean",
           "default": true
         },
+        "test_profile_volume": {
```
[the remark this answers is not in the document]
Comment (nfnt) [reply frameworks/helloworld/universe/config.json]: 👍

File: frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml [hunk 5]
```diff
@@ -14,7 +14,14 @@ pods:
       profile:
         goal: RUNNING
         essential: false
-        cmd: df -t $VOLUME_PROFILE pod-container-path >> pod-container-path/output && df -t $VOLUME_PROFILE profile-container-path >> profile-container-path/output && sleep $SLEEP_DURATION
+        cmd: >
+               set -e -x
+               [ -n "{{{TEST_PROFILE_VOLUME_COMMAND}}}" ]
```
Comment (elezar) [inline frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml]: Should this not be an `if` statement?

File: frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml [hunk 6]
```diff
@@ -28,7 +35,14 @@ pods:
       noprofile:
         goal: RUNNING
         essential: false
-        cmd: df -t $VOLUME_PROFILE pod-container-path >> pod-container-path/output && df -t ext4 noprofile-container-path >> noprofile-container-path/output && sleep $SLEEP_DURATION
+        cmd: >
+               set -e -x
+               [ -n "{{{TEST_PROFILE_VOLUME_COMMAND}}}" ]
```
Comment (elezar) [inline frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml]: Same question here? Should this be an `if` statement with the code below included in the body?

File: frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml [hunk 7]
```diff
@@ -28,7 +35,14 @@ pods:
...
+               {{{TEST_PROFILE_VOLUME_COMMAND}}} pod-container-path
+               ! {{{TEST_PROFILE_VOLUME_COMMAND}}} noprofile-container-path
```
Comment (elezar) [inline frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml]: What is the purpose of the `!` here?

File: frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml (hunk 5 above)
Comment (elezar) [reply frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml]: Sorry, just saw your comment above.

File: frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml (hunk 7 above)
Comment (elezar) [reply frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml]: Just saw your comment above.

File: frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml (hunk 6 above)
Comment (elezar) [reply frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml]: Just saw your comment above.

File: frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml (hunk 7 above)
Comment (nfnt) [reply frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml]: The `TEST_PROFILE_VOLUME_COMMAND` is used to test that a path is using a certain filesystem. For the existing test that is used to test that filesystem is using `xfs`. In the old code we were testing that `noprofile-container-path` is of type `ext4`. This doesn't work with the new approach. Hence, we're testing that we are NOT using `xfs` here.

Comment (elezar) [reply frameworks/helloworld/src/main/dist/pod-profile-mount-volume.yml]: OK, I though that `!` meant something different in bash. Thanks for the clarification.

**Pull request merged**